syntax = "proto3";

package TW.Ripple.Proto;
option java_package = "wallet.core.jni.proto";

import "Common.proto";

// Input data necessary to create a signed transaction.
message SigningInput {
    int64 amount = 1;

    int64 fee = 2;

    int32 sequence = 3;

    int32 last_ledger_sequence = 4;

    string account = 5;

    string destination = 6;

    int64 destination_tag = 7;

    int64 flags = 8;

    bytes private_key = 9;
}

// Transaction signing output.
message SigningOutput {
    bytes encoded = 1;

    // Optional error
    Common.Proto.SigningError error = 2;
}
